app/models/manageiq/providers/embedded_ansible/provider/default_ansible_objects.rb

```diff
@@ -0,0 +1,64 @@
+module ManageIQ::Providers::EmbeddedAnsible::Provider::DefaultAnsibleObjects
+  extend ActiveSupport::Concern
+
+  class DefaultAnsibleObjectError < RuntimeError; end
+
+  ANSIBLE_OBJECT_SOURCE = "MIQ_ANSIBLE".freeze
+
+  included do
+    has_many :default_ansible_objects, -> { where(:source => ANSIBLE_OBJECT_SOURCE) }, :as => :resource, :dependent => :destroy, :class_name => "CustomAttribute"
+  end
+
+  def default_organization
+    get_default_ansible_object("organization")
+  end
+
+  def default_credential
+    get_default_ansible_object("credential")
+  end
+
+  def default_inventory
+    get_default_ansible_object("inventory")
+  end
+
+  def default_host
+    get_default_ansible_object("host")
+  end
+
+  def default_organization=(org)
+    set_default_ansible_object("organization", org)
+  end
+
+  def default_credential=(cred)
+    set_default_ansible_object("credential", cred)
+  end
+
+  def default_inventory=(inv)
+    set_default_ansible_object("inventory", inv)
+  end
+
+  def default_host=(host)
+    set_default_ansible_object("host", host)
+  end
+
+  def delete_ansible_object(name)
+    default_ansible_objects.find_by(:name => name).try(:delete)
+  end
+
+  private
+
+  def get_default_ansible_object(name)
+    default_ansible_objects.find_by(:name => name).try(:value).try(:to_i)
+  end
+
+  def set_default_ansible_object(name, value)
+    current_value = get_default_ansible_object(name)
+    return if current_value == value
+
+    if current_value && current_value != value
+      raise DefaultAnsibleObjectError, "#{name} object already exists with value #{current_value}"
+    end
+
+    default_ansible_objects.create(:name => name, :value => value)
```

Comment:
All of this feels like `default_ansible_objects.find_or_initialize_by(:name => name).update_attributes(:value => value)`.  This will be a noop if the record already exists or have the same existing value.  Not sure why you need the check on lines 58-60